packages/babel-plugin-minify-dead-code-elimination/src/index.js

```diff
@@ -238,10 +247,25 @@ module.exports = ({ types: t, traverse }) => {
             }
 
             if (binding.references === 1 && binding.kind !== "param" && binding.kind !== "module" && binding.constant) {
+
               let replacement = binding.path.node;
               let replacementPath = binding.path;
+              let isAfterReturn = false;
+
               if (t.isVariableDeclarator(replacement)) {
-                replacement = replacement.init;
+
+                forEachPrevSibling(replacementPath, (path) => {
+                  if (t.isReturnStatement(path)) {
```

Comment:
Right, for each binding reference we can traverse a subtree upwards and check if parent is a previous sibling. We also need to check that identifier is not already defined:

```js
function foo() {
  ((x) => {
    return x // x - ReferencedPath but is a local var
  })() // CallExpression SiblingPath
  var x = 1;
}
```